helm/cluster-openstack/templates/_cluster.tpl

```diff
@@ -24,5 +26,7 @@ spec:
         replicas: {{ .replicas }}
         metadata:
           labels:
+            machine-deployment.giantswarm.io/failure-domain: "{{ .failureDomain }}"
```

Comment:
CAPI doesn't currently allow setting the failure domain of a MachineDeployment when using cluster topology. It does allow setting labels though. My solution is therefore to apply a label to specify the failure-domain from the node pool definition in the helm values and then allow Kyverno to default the failureDomain when CAPI creates the corresponding MachineDeployment. See https://github.com/giantswarm/kyverno-policies/pull/179